File: skills/jev-system-one/assets/wrappers/jev.ts
/**
 * One-file Jev wrapper. Copy into the product repo. Questions and thresholds
 * live in a sibling constants file (or JSON) so humans can review them.
 *
 *   npm install @typesafe-ai/sdk
 *   export TYPESAFE_API_KEY=...
 *   export JEV_ENABLED=1
 *
 * Returns { answers, model, usage, fallback: null } on success;
 * { answers: {}, model: null, usage: null, fallback: "inert"|"timeout"|errName }
 * otherwise. Callers own fail-open vs fail-closed and can stamp model/fallback
 * into telemetry (04). The client is created lazily so importing this module
 * is always inert.
 */
import { TypeSafeClient } from "@typesafe-ai/sdk";

const FLAG = process.env.JEV_ENABLED === "1";
let client: TypeSafeClient | null = null;

function getClient(): TypeSafeClient {
  client ??= new TypeSafeClient();
  return client;
}

export interface JevResult {
  answers: Record<string, unknown>;
  model: string | null;
  usage: unknown;
  fallback: string | null;
}

export async function jev(
  state: unknown,
  questions: Record<string, unknown>,
  timeoutMs = 0,
): Promise<JevResult> {
  const empty = { answers: {}, model: null, usage: null };
  if (!FLAG || !process.env.TYPESAFE_API_KEY) return { ...empty, fallback: "inert" };
  try {
    const call = getClient().systemOne({ state, questions }) as Promise<{
      answers?: Record<string, unknown>;
      model?: string;
      usage?: unknown;
    }>;
    const resp = timeoutMs
      ? await Promise.race([
          call,
          new Promise<never>((_, reject) =>
            setTimeout(() => reject(new Error("jev_timeout")), timeoutMs),
          ),
        ])
      : await call;
    return {
      answers: resp.answers ?? {},
      model: resp.model ?? null,
      usage: resp.usage ?? null,
      fallback: null,
    };
  } catch (e) {
    return {
      ...empty,
      fallback: e instanceof Error && e.message === "jev_timeout" ? "timeout" : (e as Error).name,
    };
  }
}

File: skills/jev-system-one/assets/wrappers/jev.py
"""One-file Jev wrapper. Copy into the product repo. Questions and thresholds
live in a sibling constants file (or JSON) so humans can review them.

  pip install typesafe-sdk
  export TYPESAFE_API_KEY=...
"""
from __future__ import annotations

import os
from typing import Any

from typesafe_sdk import TypeSafeClient

_CLIENT: TypeSafeClient | None = None
FLAG = os.environ.get("JEV_ENABLED", "0") == "1"


def client() -> TypeSafeClient:
    global _CLIENT
    if _CLIENT is None:
        _CLIENT = TypeSafeClient()  # reads TYPESAFE_API_KEY; default model jev-latest
    return _CLIENT


def jev(state: Any, questions: dict) -> dict:
    """Typed answers or {} on failure. Callers own fail-open vs fail-closed."""
    if not FLAG or not os.environ.get("TYPESAFE_API_KEY"):
        return {}
    try:
        return client().system_one(state=state, questions=questions).answers
    except Exception:
        return {}
